plugins/modules/foreman_architecture.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2019 Manisha Singhal (ATIX AG)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_architecture
+short_description: Manage Foreman Architectures using Foreman API
+description:
+  - Create, Update and Delete Foreman Architectures using Foreman API
+author:
+  - "Manisha Singhal (@Manisha15) ATIX AG"
+requirements:
+  - "apypie >= 0.0.1"
+options:
+  name:
+    description: Name of architecture
+    required: true
+  operatingsystems:
+    description: List of operating systems the architecture should be assigned to
+    required: false
+    default: None
+  server_url:
```

Comment:
you can drop server_url/username/password/validate_verts and just add `extends_documentation_fragment: foreman` at the end of the documentation block. see https://github.com/theforeman/foreman-ansible-modules/pull/230/